etc/testing/circle/run_hub_tests.sh

```diff
@@ -35,14 +35,27 @@ make docker-build
 make docker-push
 
 # Create a workspace running the image we just built.
-hubcli --endpoint https://hub.pachyderm.com/api/graphql --apikey $HUB_API_KEY --op create-workspace-and-wait --orgid 2193 --loglevel trace --infofile workspace.json --version $VERSION --expiration 2h --description $CIRCLE_BUILD_URL --prefix "ci-"
+for i in {1..3}; do
+	hubcli --endpoint https://hub.pachyderm.com/api/graphql --apikey $HUB_API_KEY --op create-workspace-and-wait --orgid 2193 --loglevel trace --infofile workspace.json --version $VERSION --expiration 2h --description $CIRCLE_BUILD_URL --prefix "ci-" && break
```

Comment:
Where is VERSION set.  Is this going to be the git hash?  Are the docker images already pushed at this point?

Reply:
The commands immediately before this in this script set the VERSION and build / push the images.